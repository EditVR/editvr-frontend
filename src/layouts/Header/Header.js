/**
 * @file Header.js
 * Exports EditVR's primary header.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core';

import HeaderStyles from './Header.style';
import EditVRLogo from '../../assets/editvr-logo.svg';

const Header = ({ classes }) => (
  <header className={classes.wrapper}>
    <img src={EditVRLogo} alt="EditVR logo" className={classes.logo} />
  </header>
);

Header.propTypes = {
  classes: PropTypes.shape({
    wrapper: PropTypes.string.isRequired,
    logo: PropTypes.string.isRequired
  }).isRequired
};

export default withStyles(HeaderStyles)(Header);
