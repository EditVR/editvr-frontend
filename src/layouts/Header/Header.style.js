/**
 * @file Header.style.js
 * Exports Header component styles.
 */

export default theme => ({
  wrapper: {
    animation: 'logoBackground 5s linear infinite',
    background: 'linear-gradient(to left top, cyan, magenta, yellow)',
    borderRadius: theme.spacing.unit * 3,
    display: 'flex',
    justifyContent: 'center',
    marginBottom: theme.spacing.unit * 4,
    padding: `${theme.spacing.unit * 4}px ${theme.spacing.unit * 2}px`
  },
  logo: {
    filter: 'drop-shadow( 0px 1px 8px rgba(0,0,0,0.6) )',
    maxWidth: '100%'
  }
});
