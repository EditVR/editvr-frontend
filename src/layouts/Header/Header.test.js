/**
 * @file Header.test.js
 * Contains tests for Header.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe('<Header />', () => {
  it('Matches its snapshot', () => {
    expect(renderer.create(<Header />).toJSON()).toMatchSnapshot();
  });
});
