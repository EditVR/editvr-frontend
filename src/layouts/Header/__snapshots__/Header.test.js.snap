// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Matches its snapshot 1`] = `
<header
  className="Header-wrapper-1"
>
  <img
    alt="EditVR logo"
    className="Header-logo-2"
    src="editvr-logo.svg"
  />
</header>
`;
